File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEditAddAddressEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.api.organisation.AddressDto;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditAddAddressEventDto extends OrganizationEventDto {

    private int index;
    private AddressDto address;

    OrganizationEditAddAddressEventDto() {
    }

    public OrganizationEditAddAddressEventDto(OrganisationId organisationId, int index, AddressDto address) {
        super(organisationId);
        this.index = index;
        this.address = address;
    }

    public int getIndex() {
        return index;
    }

    public AddressDto getAddress() {
        return address;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditAddAddressEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.api.organization.AddressDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditAddAddressEventDto(
    override val organizationId: OrganizationId,
    val index: Int,
    val address: AddressDto,
    override val eventApplicable: EventApplicability
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}